### ISSUE TITLE
Migrate conda packaging to conda forge

### ISSUE THREAD

**Issue by cmutel** [brightway-lca/brightway2#47]
# Migrating to conda forge

Conda forge offers a number of advantages (big team maintaining CI and packaging, easier installation), but the most important is it makes it much easier to build os- or arch-specific packages, which we absolutely need for MacOS ARM64 machines.

## Libraries

Here are the libraries which need to be added:

- [x] brightway2
- [x] brightway25
- [x] bw2analyzer
- [x] bw2calc  
- [x] bw2data
- [x] bw2io
- [x] bw2parameters
- [x] bw_migrations
- [x] bw_processing
- [x] constructive_geometries    
- [x] matrix_utils
- [x] mrio_common_metadata    
- [x] rower
- [x] stats_arrays
- [x] wurst 

For each library, [take the following steps](https://conda-forge.org/#add_recipe):

- Fork [conda forge staged-recipes](https://github.com/conda-forge/staged-recipes)
- Create a new branch from the staged-recipes master branch
- Adapt the [existing recipe](https://github.com/cmutel/conda-recipes) in a new "recipes/[your-package-name]" directory
- Propose the change as a pull request to the conda forge repo
- Ping the team when it builds correctly: ``@conda-forge/help-python, this is ready for review``

## Adapting the recipes

Make sure to add yourself as the maintainer, e.g. 

```yaml
extra:
  recipe-maintainers:
    - cmutel
```

Apparently the forge build is good enough that the following can be dropped:

```yaml
build:
  ignore_run_exports:
    - python_abi
```

Delete ``source: fn`` if present.

More complete ``test`` section:

```yaml
test:
  imports:
    - {{ name }}
  commands:
    - pip check
  requires:
    - pip
```

Change ``build: script`` command:

```yaml
build:
  script: {{ PYTHON }} -m pip install . -vv
```

Make Python version in build and run to be `>=3.6` if `noarch`

Fix license to be valid SPDX, e.g. `BSD-3-Clause`

Replace `setuptools` with `pip` in build section.

**Comment by cmutel** [comment]
- [constructive_geometries](https://github.com/conda-forge/staged-recipes/pull/19944#issuecomment-1207185126): Waiting for review
- [mrio_common_metadata](https://github.com/conda-forge/staged-recipes/pull/19945#pullrequestreview-1063863016): Waiting for review
- [stats_arrays](https://github.com/conda-forge/staged-recipes/pull/19946#issuecomment-1207180782): Waiting for review
- [bw_processing](https://github.com/conda-forge/staged-recipes/pull/19941#pullrequestreview-1063383069)

**Comment by m-rossi** [comment]
People at our company want to use packages of the brightway ecosystem. As we rely heavily on conda-forge I bring all packages we use at our company there and maintain a few feedstocks. I started creating a few recipes unitl I found this issue. Can I help you @cmutel & @michaelweinold bring the packages to conda-forge? We could also [add all packages at once](https://conda-forge.org/docs/maintainer/adding_pkgs.html#adding-multiple-packages-at-once)

**Comment by cmutel** [comment]
Dear Marco-

It would be fantastic if you could help move this process forward, there are simply too many things to do, and this requires pushing the conda-forge people a bit to get recipe submissions approved (we have waited more than a week in the past). I don't mean anything bad about the forge maintainers, they are volunteers and doing a great job, but no one has unlimited time.

I think porting the recipes should be more or less simple; what I have used in the past when building conda packages is here: https://github.com/cmutel/conda-recipes

Feel free to tag me if you have other questions.

**Comment by m-rossi** [comment]
As I said I am willing to help. How should we proceed? I see two options here:

1. Continue on the PRs you listed above. To help I need push access to your forks.
2. Start a new PR [bundling all remaining packages at once](https://conda-forge.org/docs/maintainer/adding_pkgs.html#adding-multiple-packages-at-once). We should close the other PRs then.

**Comment by cmutel** [comment]
If you can get them all done at once, that would be fantastic, and would also make it easier for the conda-forge team. I don't know anything about this, please feel free to take the lead.

**Comment by m-rossi** [comment]
Just started a PR for most of the packages listed above.

@michaelweinold can you close the PRs?

**Comment by michaelweinold** [comment]
> @michaelweinold can you close the PRs?

Done! Many thanks for your initiative.

**Comment by m-rossi** [comment]
Ok, got all recipes green and requested a review. I skipped https://github.com/conda-forge/staged-recipes/pull/20212 as it requires some development versions of packages.

If you want to help maintain the recipes please leave a comment in the PR. However I can also add you later once the PR is merged.

**Comment by m-rossi** [comment]
Just a quick update from my side: The PR with all recipes at once got rejected. So I split them up _again_. Part of them got already merged, I will update this comment with the progress:

* [ ] `activity-browser`
* [ ] `brightway2`
* [ ] `bw_migrations`: https://github.com/conda-forge/staged-recipes/pull/21372
* [ ] `bw2analyzer`: https://github.com/conda-forge/staged-recipes/pull/21373
* [x] `bw2calc`: https://github.com/conda-forge/staged-recipes/pull/21351
* [x] `bw2data`: https://github.com/conda-forge/staged-recipes/pull/21352
* [ ] `bw2io`
* [ ] `eidl`
* [ ] `presamples`: https://github.com/conda-forge/staged-recipes/pull/21375

**Comment by cmutel** [comment]
@m-rossi Thanks a lot for your persistence, this is not an easy process and I very much appreciate your work.

I agree to be a maintainer on all packages in the https://github.com/brightway-lca/ org.

**Comment by cmutel** [comment]
@m-rossi Wow, amazing. The solution to get bw2calc working on Apple Silicon is simple in the end, but took some doing! Thanks **A LOT**.

**Comment by m-rossi** [comment]
Glad I could contribute something useful and improve some part of your project. Although I am just using the amazing magic the people at conda-forge are providing. I am always amazed how such a small group can manage such a huge infrastructure and improve it everyday.

Furthermore, as I said colleagues at the company I work for are happy that a project like yours exist. If there is anything related to packaging I can help with, feel free to ping me.

**Comment by m-rossi** [comment]
I want to bring another couple of packages to conda forge.

- [ ] `wurst`
- [ ] `premise`
- [ ] `premise_gwp`

@cmutel could you close https://github.com/conda-forge/staged-recipes/pull/20035 so I can make clean start?

**Comment by cmutel** [comment]
@m-rossi Done!

**Comment by m-rossi** [comment]
Ok, there are also done.
